prow/github/types.go

```diff
@@ -551,6 +551,7 @@ type GenericCommentEvent struct {
 	IsPR         bool
 	Action       GenericCommentEventAction
 	Body         string
+	Base         string
```

Comment:
I don't think we should add this field to `GenericCommentEvent`s because it only makes sense for comments on PRs and won't be populated by `issue_comment` events on PRs.  See this thread for something similar: https://github.com/kubernetes/test-infra/pull/6738#discussion_r167123044

Reply:
Then there are two options we can have:
- either stop handling GenericCommentEvents in the approve plugin and use finer-grained handlers (more work, more prone to regress)
- or run GetPullRequest unconditionally (the only difference is that now in `pull_request_review_comments` we already have the branch as part of the webhook payload so we avoid `GetPullRequest`)

Second option sounds like the best approach since it doesn't sound we are going to save many more tokens anyway.